backend/question-service/server/app/readme.md

```diff
@@ -0,0 +1,321 @@
+    ```
+
+    ```
+    GET /questions?difficulty=Easy&difficulty=Medium&topic=Hash%20Tables&topic=Arrays # Returns all Easy or Medium questions with topic Hash Tables or Arrays
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "questions": [
+        {
+            "id": "1",
+            "title": "Find Maximum in Array",
+            ...
+        },
+        {
+            "id": "2",
+            "title": "Reverse Array",
+            ...
+        }
+        ]
+    }
+    }
+    ```
+
+2. `GET /questions/all` Get All Questions
+
+    - **Description**: Returns a list of all questions with all fields.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/all
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "questions": [
+        {
+            "id": "1",
+            "title": "Find Maximum in Array",
+            "difficulty": "Easy",
+            "topic_tags": ["Arrays"],
+            ...
+        },
+        {
+            "id": "2",
+            "title": "Reverse Array",
+            "difficulty": "Easy",
+            "topic_tags": ["Arrays"],
+            ...
+        }
+        ]
+    }
+    }
+    ```
+
+3. `GET /questions/{question_id}` Get a Single Question
+
+    - **Description**: Returns a single question matching the given id.
+    - **Path Parameters**:
+        - `question_id`: The integer ID of the question to retrieve. Required.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/1
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "question": {
+        "id": "1",
+        "title": "Find Maximum in Array",
+        "difficulty": "Easy",
+        "topic_tags": ["Arrays"],
+        ...
+        }
+    }
+    }
+    ```
+
+4. `GET /questions/filters` Get Question Filters
+
+    - **Description**: Returns a list of all possible filters for questions: Difficulty and Topic.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/filters
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "Easy": ["Arrays", "Strings"],
+        "Medium": ["Trees", "Graphs"],
+        "Hard": ["Dynamic Programming", "Recursion"]
+    }
+    }
+    ```
+
+### `/admin` Endpoints
+
+1. `POST /admin/questions` Create a New Question
+
+    - **Description**: Creates a new question in the database.
+    - **Request Body**:
+        - `question`: The question data in JSON format. **If no ID is provided, one will automatically be generated.**
+
+    **Sample Request**:
+
+    ```json
+    POST /admin/questions
+    {
+        "title": "New Question Title",
+        "difficulty": "Easy",
+        "topic_tags": ["Arrays"],
+        ...
+    }
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+        "status": "success",
+        "data": {
+            "inserted_data": {
```

Comment:
nit: same thing of renaming `inserted_data` to `question`

Reply:
Resolved in https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/commit/6617a42ae09dab7ee6beaa7d478b25451f56c033